### DISCUSSION TITLE
Digital Ocean droplet

### DISCUSSION THREAD

**Discussion by GrahamEnigma** [mastodon/mastodon#22344] [Q&A] [answered]
I added a droplet with the mastodon instance from the DO marketplace.

I have it installed (some issues with the email config, but otherwise running.  I can log in via the web browser using the admin credentials but I do not appear to have admin privileges (I get what appears to be a standard user view of preferences) certainly not config options for mastodon.

So, to problem 2. I tried ssh into the server and tried tootctl to change the account to Owner and/or Admin.  I also tried 'tootctl --version' and in each case I get this:

mastodon@mastodon:~$ cd live
mastodon@mastodon:~/live$ tootctl --version

Command 'tootctl' not found, did you mean:

  command 'bootctl' from deb systemd (245.4-4ubuntu3.19)

Try: apt install <deb name>

mastodon@mastodon:~/live$ 

I am an absolute novice at mastodon and know enough CLI to be dangerous!

Any help most welcome.

Cheers

**Comment by KahunaTheElder** [comment]
I get the same issue. I believe it is an issue with RAILS, as when I try using the CLI to make sure I have set the admin account to actually be an admin, it can't find Ruby.

**Comment by GrahamEnigma** [comment, marked as answer]
Let me know if you make progress, we seem to be the only folks interested!!!

**Reply by GrahamEnigma** [reply]
@KahunaTheElder 
I made progress on this today.  I had to do two things:
add live/bin to my path (in the mastodon user) and the set RAILS_ENV="production".  I did both in .bashrc for the mastodon user.

Then in the live directory:

rails_env=production bin/tootctl accounts modify yourusername --role Owner

That appears to have fixed it, I can now see the admin functions when I login.  I'll play with the server settings and see how far I get.

**Comment by Taiwan-SuperCat** [comment]
I have the exact same problem! 

Where do you make the changes @GrahamGIGA ? The Digital Ocean console?

**Reply by GrahamEnigma** [reply]
You need to ssh into your droplet.  Then make the changes outlined above (in my last post).  Also cracked the email problems, so now we seem (gulp) to be live.

**Reply by Taiwan-SuperCat** [reply]
Thanks a lot for your reply! 

Could you explain what is : add live/bin to my path (in the mastodon user) and the set RAILS_ENV="production". I did both in .bashrc for the mastodon user.

I keep on getting: 

Command 'tootctl' not found, did you mean:

  command 'bootctl' from deb systemd (245.4-4ubuntu3.19)

Try: apt install <deb name>

Starting a Mastodon server is a hard job! :-P

**Reply by Taiwan-SuperCat** [reply]
Someone helped me on Mastodon and it works for me :-)

Following is his suggestion: 

command line:

su - mastodon
cd /home/mastodon/live

RAILS_ENV=production bin/tootctl accounts modify [username] --role Admin

 [username] is just the local server username so you'd use bubba instead of bubba@yourserver.com

**Comment by GrahamEnigma** [comment]
You might want to get help if the instructions aren't meaningful to you. 

You need to add the location of the tootctl command (which is in live/bin) to your $PATH variable.  That tells the system, where to search for commands.  alertnatively, you can use bin/tootctl but that gets tedious, easier to add it so that it's available just by typing the command.

You need to edit .bashrc to add the RAILS_ENV variable and export it.

`su - mastodon
cd /home/mastodon/live

RAILS_ENV=production bin/tootctl accounts modify [username] --role Admin

[username] is just the local server username so you'd use bubba instead of [bubba@yourserver.com](mailto:bubba@yourserver.com)`

The 'su' command is displayed when you ssh into your droplet.  You need to do that each time you login.

you will then (already) be in the mastodon home directory.
